x-pack/apm-server/profiling/grpcext.go

```diff
@@ -42,3 +49,38 @@ func GetHostID(ctx context.Context) uint64 {
 	hostID, _ := strconv.ParseUint(hostIDs, 16, 64)
 	return hostID
 }
+
+func GetSecretToken(ctx context.Context) string {
+	md, _ := metadata.FromIncomingContext(ctx)
+	return GetFirstOrEmpty(md, MetadataKeySecretToken)
+}
+
+func GetVersion(ctx context.Context) string {
+	md, _ := metadata.FromIncomingContext(ctx)
+	return GetFirstOrEmpty(md, MetadataKeyVersion)
+}
+
+func GetRevision(ctx context.Context) string {
+	md, _ := metadata.FromIncomingContext(ctx)
+	return GetFirstOrEmpty(md, MetadataKeyRevision)
+}
+
+func GetIPAddress(ctx context.Context) string {
+	md, _ := metadata.FromIncomingContext(ctx)
+	return GetFirstOrEmpty(md, MetadataKeyIPAddress)
+}
+
+func GetHostname(ctx context.Context) string {
+	md, _ := metadata.FromIncomingContext(ctx)
+	return GetFirstOrEmpty(md, MetadataKeyHostname)
+}
+
+func GetKernelVersion(ctx context.Context) string {
+	md, _ := metadata.FromIncomingContext(ctx)
+	return GetFirstOrEmpty(md, MetadataKeyKernelVersion)
+}
+
+func GetTags(ctx context.Context) string {
+	md, _ := metadata.FromIncomingContext(ctx)
+	return GetFirstOrEmpty(md, MetadataKeyTags)
+}
```

Comment:
All these functions assume that metadata is present and `projectID` / `hostID` validated. In CA, we use an interceptor to check for this. #2620 implements this also for Elastic Collector and should be merged here too.

Reply:
Is this something that needs to happen straight away, or can it happen in a follow up? If it can wait, I'll merge this PR.

Reply:
I'll push another commit for ensuring the presence of the metadata